.github/workflows/manual-upgrade-testnet-l2.yml

```diff
@@ -0,0 +1,149 @@
+          echo "RESOURCE_TESTNET_NAME=${{env.RESOURCE_TESTNET_NAME}}" >> $GITHUB_OUTPUT
+
+      - name: 'Login to Azure docker registry'
+        uses: azure/docker-login@v1
+        with:
+          login-server: testnetobscuronet.azurecr.io
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: 'Build and push obscuro node images'
+        run: |
+          docker build -t ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}} -f dockerfiles/enclave.Dockerfile  .
+          docker push ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}} -f dockerfiles/host.Dockerfile .
+          docker push ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG}} -f testnet/contractdeployer.Dockerfile .
+          docker push ${{env.L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG}} -f testnet/hardhatdeployer.Dockerfile .
+          docker push ${{env.L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG}}
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        host_id: [ 0,1 ]
+
+    steps:
+      - name: 'Extract branch name'
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF_NAME})"
+        id: extract_branch
+
+      - name: 'Login via Azure CLI'
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: 'Update Obscuro node-${{ matrix.host_id }} on Azure'
+        uses: azure/CLI@v1
+        with:
+          inlineScript: |
+            az vm run-command invoke -g Testnet -n "${{needs.build.outputs.RESOURCE_STARTING_NAME}}-${{ matrix.host_id }}-${{ github.event.inputs.github_deploy_number }}"  \
+            --command-id RunShellScript \
+            --scripts '
+              cd /home/obscuro/go-obscuro/testnet/ \
+              docker compose down host enclave && \
+              docker compose pull && \
+              docker compose up host enclave
+            '
+
+  wait-node-up:
+    needs:
+      - build
+      - deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: "Wait until obscuro node is healthy"
+        shell: bash
+        run: |
+          ./.github/workflows/runner-scripts/wait-node-healthy.sh --host=obscuronode-0-${{needs.build.outputs.RESOURCE_TESTNET_NAME}}-${{ github.event.inputs.github_deploy_number }}.uksouth.cloudapp.azure.com
+
+  deploy-faucet:
+    runs-on: ubuntu-latest
+    needs: wait-node-up
+    steps:
+      - name: 'Trigger Faucet deployment workflow'
+        run: |
+          curl -XPOST -H "Authorization: Bearer ${{ secrets.GH_TOKEN }}" -H "Accept:application/vnd.github" -H "Content-Type:application/json" https://api.github.com/repos/obscuronet/faucet/actions/workflows/manual-deploy-testnet-faucet.yml/dispatches --data '{"ref": "main" }'
```

Comment:
How does this differentiate between testnet and dev-testnet? Pretty sure this will always kick off the testnet faucet deployment even when dev-testnet chosen as the testnet_type.

Reply:
It shouldn't deploy the faucet again, if the blockchain state is accurate then restarting the nodes will not impact the faucet.